api/src/pcapi/core/users/email/update.py

```diff
@@ -165,14 +163,18 @@ def validate_email_update_request(
     Therefore this function can be called multiple times with the same
     inputs safely.
     """
-    current_user = users_repository.find_user_by_email(current_email)
-
-    if not current_user:
+    payload = ChangeEmailTokenContent.from_token(token)
+    current_email = payload.current_email
+    new_email = payload.new_email
+    user = users_repository.find_user_by_email(current_email)
+    if not user:
         if not validated_update_request_exists(current_email, new_email):
             raise exceptions.UserDoesNotExist()
-    else:
-        api.change_email(current_user, new_email, by_admin)
-        transactional_mails.send_email_change_information_email(current_user)
+        return
+    check_and_expire_token(user, token, TokenType.VALIDATION)  # TODO
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
**💣 chore:** ‏TODO ?